### ISSUE TITLE
Component that returns an array is not recognised

### ISSUE THREAD

**Issue by lensbart** [styleguidist/react-styleguidist#1365]
**Current behavior**

Components that return an array lead to the following error:

`Warning: some/folder/index.jsx matches a pattern defined in “components” or “sections” options in your style guide config but doesn’t export a component.`

`It usually happens when using third-party libraries, see possible solutions here:
https://react-styleguidist.js.org/docs/thirdparties.html`

**To reproduce**

```jsx
/**
 * @param {Props} props props
 * @property {Array.<Etcetera>} etc
 * @returns {React.Node} ...
 */
const Contents = ({ contents }: Props) =>
	contents.map(data => {
		switch (data.__typename) {
			case 'Something':
				return (
					<Something children={data.children} key={data.id} />
				)
		}
	})
```

**Expected behavior**

No warning

**Comment by sapegin** [comment]
Likely should be reported at https://github.com/reactjs/react-docgen

**Comment by lensbart** [comment]
Okay, I posted it there. Thanks
